### DISCUSSION TITLE
clog-web: how to reload content?

### DISCUSSION THREAD

**Discussion by vindarel** [rabbibotton/clog#80] [Q&A] [answered]
Hello there!

I am trying the 24th tutorial about clog-web. I run `start-tutorial` and see the content, then I edit a div (create-section, below "try and adjust…"), I compile the function, refresh the browser and I don't see the new content. I was expecting to. If I re-run `start-tutorial` then I see it again.

Is it normal behavior? It would be great to not have to re-run start-tutorial everytime. 

*(Would this be a Clack limitation? (I read a similar complain for Caveman. When I use Hunchentoot I can re-complie a route definition, refresh the page and see the changes, without the need to restart the server).)*

**Comment by aykaramba** [comment]
I would like to know the answer to this as well.

However, because I am learning lisp by using clog, what I have started to do is to run (clog-repl) and re-create the tutorials in the repl.   The benefit of this is that you see the updates immediately and it forces you to chop up the tutorial forms into bits and pieces and try permutations until you can see how everything works.

I am immensely grateful that the author of this framework wrote a bunch of tutorials. The manual is incredibly useful as well.

Hopefully we can figure out how the live coding workflow needs to operate for real projects.

**Reply by smvamse** [reply]
The `on-new-window` handler should be a symbol eg. `'on-new-window` instead of a function object eg. `#'on-new-window`

**Reply by aykaramba** [reply]
Thanks!

**Comment by rabbibotton** [comment, marked as answer]
When you recompile the on-new-window handler you need to set it again with set-on-new-window as the path is still pointing to the old compiled code. (set-on-new-window #'on-new-window)

Alternatively you can use (set-on-new-window 'on-new-window) ie using the symbol instead of the function symbol and you no longer have to set each time the handler can do that also in INITIALIZE.

Another alternative is to not place the code to write the page in your on-new-window-handler but have the on-new-window-handler call another function and recompile that function instead.

**Reply by aykaramba** [reply]
Awesome.  Thanks!

**Reply by vindarel** [reply]
Oh yes, great. 

So now my `on-new-window` is simply this:

```lisp
(defun on-new-window (body)
  (clog-web-initialize body)
  (make-my-window body))
```

and `make-my-window` is all the rest.
